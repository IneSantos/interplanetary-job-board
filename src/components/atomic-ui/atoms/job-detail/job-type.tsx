export const JobDetailsEnum = {
	COMPANY: "company",
	LOCATION: "location",
	CONTRACT: "contract",
	NONE: "none"
}