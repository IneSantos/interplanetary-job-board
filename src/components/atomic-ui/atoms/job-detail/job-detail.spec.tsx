import React from "react";
import { render } from '@testing-library/react';
import { JobDetailsEnum } from "./job-type";
import JobDetail from "./job-detail";

describe("JobDescription atom tests", () => {
  test(`JobDescription component render COMPANY`, () => {
    const { container } = render(
      <JobDetail type={JobDetailsEnum.COMPANY} description="This is a description"/>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test(`JobDescription component render LOCATION`, () => {
    const { container } = render(
      <JobDetail type={JobDetailsEnum.LOCATION} description="This is a description"/>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test(`JobDescription component render CONTRACT`, () => {
    const { container } = render(
      <JobDetail type={JobDetailsEnum.CONTRACT} description="This is a description"/>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test(`JobDescription component render NONE`, () => {
    const { container } = render(
      <JobDetail type={JobDetailsEnum.NONE} description="This is a description"/>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  
});
