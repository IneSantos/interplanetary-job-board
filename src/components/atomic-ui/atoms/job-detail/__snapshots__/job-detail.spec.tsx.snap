// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobDescription atom tests JobDescription component render COMPANY 1`] = `
<div
  class="job-detail job-detail-company"
>
  <i
    class="fas fa-building icon"
  />
  <span>
    This is a description
  </span>
</div>
`;

exports[`JobDescription atom tests JobDescription component render CONTRACT 1`] = `
<div
  class="job-detail job-detail-contract"
>
  <i
    class="fas fa-file-signature icon"
  />
  <span>
    This is a description
  </span>
</div>
`;

exports[`JobDescription atom tests JobDescription component render LOCATION 1`] = `
<div
  class="job-detail job-detail-location"
>
  <i
    class="fas fa-map-marker-alt icon"
  />
  <span>
    This is a description
  </span>
</div>
`;

exports[`JobDescription atom tests JobDescription component render NONE 1`] = `
<div
  class="job-detail job-detail-none"
>
  <span>
    This is a description
  </span>
</div>
`;
